code/clue.js

```diff
@@ -1,59 +1,274 @@
+};
+
+const toggleAnimationDeckCard = (id) => {
+  document.getElementById(id).classList.toggle("deck-heartbeat");
+};
+
+// Pick killer function
 const pickKiller = () => {
-  // This will randomly select a killer from the suspects. And add that to the mystery object.
-  mystery.killer = randomSelector(suspects)
+  if (!killerPicked) {
+    cleanRevealMysteryMessage(); // Removes "Reveal the crime" message if any
+    toggleAnimationDeckCard("killerCard");
+    setTimeout(() => {
+      shuffleFavouriteWeapon();
+      mystery.killer = randomSelector(suspects);
+      document.getElementById("killerCard").style.background =
+        mystery.killer.color;
+      document.getElementById(
+        "killerName"
+      ).innerHTML = `${mystery.killer.firstName} ${mystery.killer.lastName}`;
+      document.getElementById(
+        "killerAge"
+      ).innerHTML = `Age: ${mystery.killer.age}`;
+      document.getElementById(
+        "killerOccupation"
+      ).innerHTML = `Occupation: ${mystery.killer.occupation}`;
+      document.getElementById("killerImage").src = `${mystery.killer.image}`;
+      document.getElementById(
+        "killerDescription"
+      ).innerHTML = `${mystery.killer.description}`;
+      document.getElementById(
+        "killerFavouriteWeapon"
+      ).innerHTML = `Favourite Weapon: ${mystery.killer.favouriteWeapon}`;
+    }, 2000);
+    killerPicked = true;
+  }
+};
+
+// Pick weapon function
+const pickWeapon = () => {
+  if (!weaponPicked) {
+    cleanRevealMysteryMessage(); // Removes "Reveal the crime" message if any
+    toggleAnimationDeckCard("weaponCard");
+    setTimeout(() => {
+      mystery.weapon = randomSelector(weapons);
+      document.getElementById(
+        "weaponName"
+      ).innerHTML = `${mystery.weapon.name}`;
+      document.getElementById(
+        "weaponWeight"
+      ).innerHTML = `Weight: ${mystery.weapon.weight} lbs`;
+    }, 2000);
+    weaponPicked = true;
+  }
+};
 
-  // This will change the background color of the card to the one connected to the chosen killer and show the full name of the killer. Feel free to add more things to show about the killer.
-  document.getElementById('killerCard').style.background = mystery.killer.color
-  document.getElementById(
-    'killerName'
-  ).innerHTML = `${mystery.killer.firstName} ${mystery.killer.lastName}`
-}
+// Pick room function
+const pickRoom = () => {
+  if (!roomPicked) {
+    cleanRevealMysteryMessage(); // Removes "Reveal the crime" message if any
+    toggleAnimationDeckCard("roomCard");
+    setTimeout(() => {
+      mystery.room = randomSelector(rooms);
+      document.getElementById("roomName").innerHTML = `${mystery.room}`;
+    }, 2000);
+    roomPicked = true;
+  }
+};
 
-// CREATE FUNCTIONS pickWeapon and pickRoom in a similar way.
+// Reveal function
+const revealMystery = () => {
+  if (
+    mystery.killer === null ||
+    mystery.weapon === null ||
+    mystery.room === null
```

Comment:
You can simplify it :

```
  if (!mystery.killer || !mystery.weapon || !mystery.room) { ... }
```

for more information, check out https://www.freecodecamp.org/news/js-type-coercion-explained-27ba3d9a2839/